gravitee-apim-console-webui/src/portal/navigation-items/portal-navigation-items.component.ts

```diff
@@ -439,6 +440,65 @@ export class PortalNavigationItemsComponent {
       )
       .subscribe();
   }
+
+  onNodeMoved($event: NodeMovedEvent) {
+    const { node, newParentId, newOrder } = $event;
+
+    if (this.contentControl.pristine) {
+      this.updatePortalNavigationItemOrderAndRefreshList(newParentId, newOrder, node.data)
+        .pipe(takeUntilDestroyed(this.destroyRef))
+        .subscribe();
+    } else {
+      this.openDiscardChangesDialog()
+        .pipe(
+          filter((confirmed) => !!confirmed),
+          switchMap(() => this.updatePortalNavigationItemOrderAndRefreshList(newParentId, newOrder, node.data)),
+          takeUntilDestroyed(this.destroyRef),
+        )
+        .subscribe();
+    }
+  }
+
+  private openDiscardChangesDialog() {
+    return this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData, boolean>(GioConfirmDialogComponent, {
+        width: GIO_DIALOG_WIDTH.SMALL,
+        data: {
+          title: 'Discard changes',
+          content: 'There are unsaved changes on this page. Would you like to discard changes or continue editing?',
+          confirmButton: 'Discard Changes',
+          cancelButton: 'Continue Editing',
+        },
+        role: 'alertdialog',
+        id: 'discardChangesConfirmDialog',
+      })
+      .afterClosed();
+  }
+
+  private updatePortalNavigationItemOrderAndRefreshList(
```

Comment:
Maybe shorten the name a little bit

```suggestion
  private updateItemOrderAndRefreshList(

Reply:
Done!